library/schema_check.py

```diff
@@ -79,9 +79,9 @@ def check_file_paths(module, *args):
 
 def validate_grp_names(data):
     res_grps = data['resource_groups']
-    res_grp_vars = data['resource_group_vars']
+    res_grp_vars = data['resource_group_vars'] if 'resource_group_vars' in data.keys() else []
```

Comment:
I'd highly recommend list comprehensions here. It's cleaner.

http://stackoverflow.com/questions/4406389/if-else-in-a-list-comprehension